### PR TITLE
Mockdevice thermometer

### PR DESCRIPTION
**Pull request by Twpeak**
**What this PR does / why we need it**:

**Will this PR make the community happier**? 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [x] unit test
- [ ] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer.go [hunk 1]
```diff
@@ -27,10 +27,10 @@ func instructionHandler(functionName string) http.HandlerFunc {
 			rand.Seed(time.Now().UnixNano())
 			min := 10
 			max := 30
-			fmt.Fprintln(w, strconv.Itoa(rand.Intn(max-min+1)+min))
+			fmt.Fprint(w, strconv.Itoa(rand.Intn(max-min+1)+min))
```
Comment (kris21he) [inline pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer.go:30]: why not Fprintln?

File: pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	os.Setenv("MOCKDEVICE_NAME", "mockdevice_test")
+	os.Setenv("MOCKDEVICE_PORT", "12345")
+	mocks := []struct {
+		name       string
+		url        string
+		StatusCode int
+		expResult  interface{}
+	}{
+		{
+			"case 1 prot 12345 read_value",
```
Comment (kris21he) [inline pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer_test.go:29]: prot to "port"

File: pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		t.Run(c.name, func(t *testing.T) {
+			resp, err := http.Get(c.url)
+			if err != nil {
+				t.Fatalf("HTTP GET returns an error %v", err.Error())
+			}
+			defer resp.Body.Close()
+			assert.Nil(t, err)
+			body, _ := io.ReadAll(resp.Body)
+
+			if c.name == mocks[len(mocks)-1].name {
```
Comment (kris21he) [inline pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer_test.go:56]: try not use case name to switch the testing code, if the cases are different, we can avoid tabling the case. just write them strait forward

File: pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer.go (hunk 1 above)
Comment (Twpeak) [reply pkg/deviceshifu/mockdevice/mockdevice-thermometer/mockdevice-thermometer.go:30]: Because the result of 'Fprintln' ends with \n

**Review by kris21he — APPROVED** [review]

**Pull request merged**